### DISCUSSION TITLE
root_url meaning and usage

### DISCUSSION THREAD

**Discussion by jpassaro** [Homebrew/discussions#3014] [Tap maintenance and Homebrew development]
I'm maintaining a homebrew tap in my org's private github. It would be nice to generate bottles for the formulae. I've gotten as far as generating the bottle tar and json metadata, but hosting it on the usual "root url" is not an option. We have an artifactory instance which could be appropriate for hosting the bottles or we could use github releases, but I'm not sure how to configure the formula to accommodate either of those possibilities.

One thing that could help is more information about the `root_url` parameter in the formula's `bottle` section. How is it used -- i.e. how do you get from the root url, the formula details, and your current OS version, to a bottle download URL? [This section on the homebrew website](https://docs.brew.sh/Bottles#root-url-root_url) is a bit sparse.

Is it as simple as `<root url>/<bottle file name>`, e.g. `<root_url>/my-formula--0.0.1.catalina.bottle.2.tar.gz`? 

If this is accurate I'd be happy to incorporate it and submit a PR.

**Comment by SMillerDev** [comment]
You can use `brew tap-new` to set up a tap with a github bottles action pre-configured. That very likely the easiest way.
